src/port.js

```diff
@@ -265,7 +265,7 @@ export default class Port {
 	}
 	static validateEvent(targetOrigin, endpoint, e) {
 		var isValid = (e.source === endpoint) &&
-			(targetOrigin === '*' || targetOrigin === e.origin) &&
+			(targetOrigin === '*' || targetOrigin != null && e.origin != null && targetOrigin.toUpperCase() === e.origin.toUpperCase()) &&
```

Comment:
You'll want a typesafe comparison here: `targetOrigin !== null && e.origin !== null`